ratpack-rx2/src/main/java/ratpack/rx2/RxRatpack.java

```diff
@@ -0,0 +1,729 @@
+    );
+  }
+
+
+  /**
+   * Converts a {@link Operation} into a {@link Completable}.
+   * <p>
+   * The returned Completable emits nothing if it succeeds, and emits the error (i.e. via {@code onError()}) if it fails.
+   */
+  public static Completable complete(Operation operation) {
+    return Completable.create(subscriber ->
+      operation.onError(subscriber::onError).then(subscriber::onComplete)
+    );
+  }
+
+  /**
+   * Observe a promise list as an observable stream.
+   *
+   * @param promise
+   * @param <T>
+   * @return
+   */
+  public static <T, I extends Iterable<T>> Observable<T> observe(Promise<I> promise) {
+    return Observable.merge(single(promise).toObservable().map(Observable::fromIterable));
+  }
+
+  /**
+   * Converts an {@link Observable} into a {@link Promise}, for all of the observable's items.
+   * <p>
+   * This method can be used to simply adapt an observable to a promise, but can also be used to bind an observable to the current execution.
+   * It is sometimes more convenient to use {@link #promiseAll(ObservableOnSubscribe)} over this method.
+   * <p>
+   * <pre class="java">{@code
+   * import ratpack.rx2.RxRatpack;
+   * import ratpack.test.exec.ExecHarness;
+   * import io.reactivex.Observable;
+   * import java.util.List;
+   * import java.util.Arrays;
+   * import static org.junit.Assert.assertEquals;
+   *
+   * public class Example {
+   *   public static class AsyncService {
+   *     public <T> Observable<T> observe(final T value) {
+   *       return Observable.create(subscriber ->
+   *         new Thread(() -> {
+   *           subscriber.onNext(value);
+   *           subscriber.onComplete();
+   *         }).start()
+   *       );
+   *     }
+   *   }
+   *
+   *   public static void main(String[] args) throws Throwable {
+   *     List<String> results = ExecHarness.yieldSingle(execution ->
+   *       RxRatpack.promiseAll(new AsyncService().observe("foo"))
+   *     ).getValue();
+   *
+   *     assertEquals(Arrays.asList("foo"), results);
+   *   }
+   * }
+   * }</pre>
+   * <p>
+   * This method uses {@link Observable#toList()} to collect the observable's contents into a list.
+   * It therefore should not be used with observables with many or infinite items.
+   * <p>
+   * If it is expected that the observable only emits one element, it is typically more convenient to use {@link #promise(Single)}.
+   * <p>
+   * If the observable emits an error, the returned promise will fail with that error.
+   * <p>
+   * This method must be called during an execution.
+   *
+   * @param observable the observable
+   * @param <T>        the type of the value observed
+   * @return a promise that returns all values from the observable
+   * @throws UnmanagedThreadException if called outside of an execution
+   * @see #promise(Single)
+   */
+  public static <T> Promise<List<T>> promiseAll(Observable<T> observable) throws UnmanagedThreadException {
```

Comment:
Given this does the expensive toList I think its good to have a slightly different name.